File: solutions/problem_1513_number_of_substrings_with_only_1s/solution.py
"""
Problem 1513: Number of Substrings With Only 1s
=============================================
Difficulty: Medium
Tags: Math, String
Pattern: Math

Time Complexity:  O(n) or O(sqrt(n))
Space Complexity: O(1)
"""

class Solution:
    def numSub(self, s: str) -> int:
        # Mathematical approach
        result = 0
        x = s
        while x != 0:
            result = result * 10 + x % 10
            x //= 10 if isinstance(x, int) else 1
        return result

File: solutions/problem_1513_number_of_substrings_with_only_1s/solution.cpp
/*
 * Problem 1513: Number of Substrings With Only 1s
 * =============================================
 * Difficulty: Medium
 * Tags: Math, String
 * Pattern: Math
 *
 * Time Complexity:  O(n) or O(sqrt(n))
 * Space Complexity: O(1)
 */

#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int numSub(string& s) {
        // Mathematical approach
        long long result = 0;
        int x = s;
        while (x != 0) {
            result = result * 10 + x % 10;
            x /= 10;
        }
        return (int)result;
    }
};
